nil/core-concepts/contract-co-location.mdx

```diff
@@ -0,0 +1,118 @@
+# Contract co-location
+
+## Definition
+
+[**Cross-shard communications**](./shards-parallel-execution#message-passing-checks) are a built-in feature of the =nil; protocol ensuring horizontal scalability with no fragmentation.
+
+However, =nil; also supports contract co-location, which is a technique for ensuring that two accounts are located on the same shard. In this case, interaction between these accounts does not require cross-shard communications.
```

Comment:
In my opinion, since we've changed the sharding model, co-location term doesn't make a lot of sense. But I'd like to hear @AviZurlo opinion on this